modules/publication/php/publication.class.inc

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * Publications module class
+ *
+ * PHP version 7
+ *
+ * @category Behavioural
+ * @package  Main
+ * @author   David Blader  <dblader.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/mohadesz/Loris-Trunk
+ */
+namespace LORIS\publication;
+
+/**
+ * Module for viewing research project proposals and their respective publications
+ *
+ * @category Behavioural
+ * @package  Main
+ * @author   David Blader  <dblader.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+class Publication extends \NDB_Menu_Filter_Form
+{
+    public $AjaxModule   = true;
+    public $skipTemplate = true;
+    /**
+     * The _has_access returns true
+     * if the user has the specific permission
+     *
+     * @return boolean
+     */
+    function _hasAccess()
+    {
+        $user = \User::singleton();
+        $db   = \Database::singleton();
+
+        $nProjects = $db->pselectOne(
```

Comment:
Is this part necessary? Will certain editors be in `publication_users_edit_perm_rel` and also lack all of the other 3 permissions?

Reply:
yes, the idea is that people can allow whomever to edit a project they propose without having to ask project administrators to grant permissions every time they want someone to maintain / collaborate on a project with them